libs/tree-sitter-wing/grammar.js

```diff
@@ -502,6 +503,23 @@ module.exports = grammar({
     await_expression: ($) => prec.right(seq("await", $.expression)),
     defer_expression: ($) => prec.right(seq("defer", $.expression)),
     parenthesized_expression: ($) => seq("(", $.expression, ")"),
+    
+    json_literal: ($) => choice(
+      seq(choice("Json", "JsonMut"), field("element", $._json_element)),
+    ),
+    _json_element: ($) => choice(
+      $.json_map_literal,
+      $.json_array_literal,
+      $.string,
+      $.number,
+      $.bool
+    ),
+    json_array_literal: ($) => seq(
+     "[", commaSep(field("element", $.expression)), "]"
+    ),
+    json_map_literal: ($) => seq(
+     "{", commaSep(field("member", $.map_literal_member)), "}"
```

Comment:
The problem with this is that nested maps within Json will not be Json map literals and not persist the heterogenous behavior. 

I could keep going and make a json_map_literal_member but then I feel like Im basically copying the whole grammar into json versions

Reply:
I hope I understand you correctly, but if you need to reuse logic from map or any other element of the grammar, you can easily do so, since this is javascript. See for example the `commaSep()` function at the bottom of the file.

To get help, you can schedule some time with @MarkMcCulloh, our master of grammar 😄

Reply:
As unfortunate as it is, I think we may have to deal with creating json-specific grammar nodes. Even though most of the json stuff is super similar to existing rules it's not exactly the same. 
I don't think our grammar will makes it feasible to allow the rest of our `expressions` inside json literals (due to the inconsistency) so it'll have to be limited.

I think it would work to have:
- json_element (any of the following)
  - json_primitive (`string|number|boolean|identifier`)
  - json_map (`{identifier|string: json_element}`)
  - json_array (`[json_element]`)